### DISCUSSION TITLE
Questions on NFT transferability

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#669] [General]
Questions from @smol-ninja:

1. Can NFT transferability only be toggled by sender?
2. When transferability is disabled, how should it be dealt in case of `withdrawMaxAndTransfer()`? This function transfers the NFT to a new recipient and is called by `Stream` owner.
3. Should the default case be transfer enabled?

**Comment by PaulRBerg** [comment]
1. Yes, but there's more nuance. Please take a look at my comment below.
2. `withdrawMaxAndTransfer` should be disabled, too.
3. There is no such thing as a "default case" at the contract level, only at the frontend level. All create `Create` function parameters must be updated to take a new boolean. And function params cannot take default values in Solidity.

---

I think you meant to ask, "Should we have a granular access control mechanism that allows the stream creator to disallow either the sender or the recipient from transferring the stream?". The answer to this question is "no".

Today, the NFT can be transferred only by the recipient. The goal with https://github.com/sablier-labs/v2-core/issues/630 is to turn off the transferability for recipients, not to make it possible for the sender to transfer the NFT on behalf of the recipient.

As a side note, giving permissions to the sender can be problematic; see https://github.com/sablier-labs/v2-core/discussions/666.

**Comment by smol-ninja** [comment]
Thanks for your comments. I think my questions were not appropriate but some of your comments are very useful. Let me try to rephrase them. 

Questions:
1. Do I understand it correctly when I say there are two parties involved- stream creator and stream recipient (who is also the NFT owner); only stream recipient can transfer NFT to a new recipient in the current design while stream creator creates the stream after setting parameters in the UI; & in the new design only stream creator will set `transferrable` parameter for the NFT? If the `transferrable` is `true`, the stream recipient can transfer NFT to another address but if its `false`, the stream recipient cannot transfer NFT to any address?

2. Should the stream creator set transferability parameter at the time of stream creation or should the creator be allowed to change it later as well, assuming the stream is active? For instance, stream creator chooses to disable it in the beginning and may have reasons to enable it again in the future.

3. When the NFT transferability is set to `false`, should depleted NFTs be allowed to be burnt by the stream recipient or not? We can allow this by checking if the new recipient is `0x0` address.

Also my question about the default value for the `transferrable` param was purely for the purpose of test cases (to be set in the `Defaults.sol`). I understand I cannot set default value for parameters in Solidity.

**Reply by PaulRBerg** [reply]
1. Two points:
    - Strictly speaking, there are three parties involved: funder, sender, and recipient (see the definition of the [`CreateLockupLinearStream`](https://github.com/sablier-labs/v2-core/blob/925630dd28f4a47ff2883b281af812f9958b957e/src/interfaces/ISablierV2LockupLinear.sol#L28C11-L38) event). But in the vast majority of cases, the funder is the same as the sender, so we can speak of either of these parties as the "stream creator".
    - The rest of your understanding is correct.
2. Let's have the first implementation be as minimalistic as possible. No ability to disable or enable transferability once the stream is created.
3. Yes, burns should be allowed.

> my question about the default value for the transferrable param was purely for the purpose of test cases (to be set in the Defaults.sol)

Ah, I see. Let's default to making the stream transferrable.
